src/Controller/DocumentController.php

```diff
@@ -145,13 +150,20 @@ public function create(CaseEntity $case, Request $request): Response
      * @throws FileMovingException
      * @throws DocumentDirectoryException
      */
-    public function edit(CaseEntity $case, Document $document, Request $request): Response
+    public function edit(CaseEntity $case, Document $document, CaseEventHelper $caseEventHelper, Request $request): Response
     {
         $this->denyAccessUnlessGranted('edit', $case);
-        $form = $this->createForm(DocumentType::class, $document);
+        $form = $this->createForm(DocumentType::class, $document, [
+            'case' => $case,
+            'view_timezone' => $this->serviceOptions['view_timezone'],
+        ]);
 
         $form->handleRequest($request);
         if ($form->isSubmitted() && $form->isValid()) {
+            if (DocumentType::CASE_EVENT_OPTION_YES === $form->get('createCaseEvent')->getData()) {
+                $this->handleCaseEventCreation($case, [$document], $form->get('caseEvent'));
+            }
```

Comment:
This a almost the exact same code as in lines 131–133. Consider just passing documents and the form to `handleCaseEventCreation` and let it check if a case event should be created and do it if so.

Reply:
Good point. Updated.